cep-20.2.md

```diff
@@ -0,0 +1,727 @@
+# A new recipe format – part 2 - the allowed keys & values
+
+<table>
+<tr><td> Title </td><td> A new recipe format - values </td>
+<tr><td> Status </td><td> Proposed</td></tr>
+<tr><td> Author(s) </td><td> Wolf Vollprecht &lt;wolf@prefix.dev&gt;</td></tr>
+<tr><td> Created </td><td> May 23, 2023</td></tr>
+<tr><td> Updated </td><td> May 23, 2023</td></tr>
+<tr><td> Discussion </td><td>  </td></tr>
+<tr><td> Implementation </td><td>https://github.com/prefix-dev/rattler-build</td></tr>
+</table>
+
+## Abstract
+
+We propose a new recipe format that is heavily inspired by conda-build. The main
+change is a pure YAML format without arbitrary Jinja or comments with semantic
+meaning.
+
+## Motivation
+
+The conda-build format is currently under-specified. For the new format, we try to list all values
+and types in a single document and document them. This is part of a larger effort (see [](/cep-20.1.md) for the new YAML syntax).
+
+## History
+
+A discussion was started on what a new recipe spec could or should look like.
+The fragments of this discussion can be found here:
+https://github.com/mamba-org/conda-specs/blob/master/proposed_specs/recipe.md
+The reason for a new spec are:
+
+- Make it easier to parse ("pure yaml"). conda-build uses a mix of comments and
+  jinja to achieve a great deal of flexibility, but it's hard to parse the
+  recipe with a computer
+- iron out some inconsistencies around multiple outputs (build vs. build/script
+  and more)
+- remove any need for recursive parsing & solving
+- cater to needs for automation and dependency tree analysis via a determinstic
+  format
+
+## Major differences with conda-build
+
+Outputs and top-level package are mutually exclusive, and outputs have exactly the same structure as top-level keys.
+If outputs exist, the top-level keys are 'merged' with the output keys (e.g. for the about section).
+
+# Format
+
+## Schema version
+
+The implicit version of the YAML schema for a recipe is an integer 1. 
+To discern between the "old" format and the new format, we utilize the file name.
+The old format is `meta.yaml` and the new format is `recipe.yaml`.
+The version can be explicitly set by adding a `schema_version` key to the recipe.
+
+```yaml
+# optional, since implicitly defaults to 1
+schema_version: 1  # integer
+```
```

Comment:
I wouldn't force a rename of `meta.yaml` (but I'd welcome the option to calling it `recipe.yaml`). Instead, I'd enforce `schema_version>=1` for the newer format, so no `schema_version` implies `0` and is today's underspecified format.

Reply:
The problem with that is that the old format is not valid yaml. So to properly know, one would have to use some of the old parsing logic/magic as well.